### PR TITLE
test: migrate to vitest

### PR DESCRIPTION
**Pull request by merceyz**
Reduces the amount of configuration and dependencies needed to run the tests with almost no changes to the code.

Ref https://github.com/nodejs/corepack/pull/229#discussion_r1064149939

### CODE REVIEW

**Comment by aduh95** [comment]
I'm -0.5 on this, Vitest shares a lot of the same problems as Jest (you're not testing on Node.js, you're testing in Jest/Vitest runtime), I would rather like to see us move away from this and use `node:test` or Mocha instead.

**Comment by merceyz** [comment]
Indeed, nothing changes in that regard here, though to be fair most of the tests spawn Corepack in a `node` process so most of the work happens outside of the Jest/Vitest runtime.

This PR landing doesn't prevent us from migrating to `node:test` in the future, this PR could land and a follow-up PR could migrate to `node:test`.

**Comment by aduh95** [comment]
> most of the tests spawn Corepack in a node process so most of the work happens outside of the Jest/Vitest runtime.

Yes that’s true, I did not consider that. 

It seems there are some changes which are not directly related to switching to Vitest in this PR, right? Do you know why the nock files need to change?

**Comment by merceyz** [comment]
Most of them were renamed because of https://github.com/nodejs/corepack/pull/349/files#diff-b017716ca1490209cba877efb506d6b0cd9d724dda50f33a7384a88da852067fR23
```diff
- http utils fetchUrlStream rejects with error
+ tests/httpUtils.test.ts > http utils fetchUrlStream > rejects with error
```

The two that actually changed deserialises to the same content.

File: .yarn/patches/clipanion-npm-3.2.1-fc9187f56c.patch
Comment (merceyz) [inline .yarn/patches/clipanion-npm-3.2.1-fc9187f56c.patch:1]: @arcanis Clipanion could probably use the `imports` field for this.

Comment (merceyz) [reply .yarn/patches/clipanion-npm-3.2.1-fc9187f56c.patch:1]: Upstream PR https://github.com/arcanis/clipanion/pull/155

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,5 @@
 
 /shims
 /coverage
+
+/node_modules
```
Comment (aduh95) [inline .gitignore:14]: Why this change?

Comment (merceyz) [reply .gitignore:14]: Vitest stores data in `node_modules` so I ignored it.

```console
$ find node_modules/
node_modules/
node_modules/.vitest
node_modules/.vitest/deps
node_modules/.vitest/deps/_metadata.json
node_modules/.vitest/deps/package.json
node_modules/.vitest/results.json
```

**Comment by merceyz** [comment]
> I would rather like to see us move away from this and use node:test or Mocha instead.

I tested using `node:test` (ref https://github.com/nodejs/corepack/pull/357) but didn't finish it.

**Comment by wojtekmaj** [comment]
While concerns about Jest/Vitest shared here may be valid, migration to Vitest will bring a clear improvement of our dev environment, and will enable me to proceed with more modernization actions that I have on my radar.

**Review by aduh95 — APPROVED** [review]
Can you please fix the conflicts?

File: package.json [hunk 2]
```diff
@@ -47,6 +40,7 @@
     "typescript": "^5.3.3",
     "undici": "^6.19.2",
     "v8-compile-cache": "^2.3.0",
+    "vitest": "^1.3.1",
```
[the remark this answers is not in the document]
Comment (merceyz) [reply package.json:43]: It was released 4 days ago, I'd rather wait a while to make sure most of the initial bugs are fixed first.

[the remark this answers is not in the document]
Comment (aduh95) [reply package.json:43]: Let's try to update it, and as long as CI's happy, it means those initial bugs do not concern us

[the remark this answers is not in the document]
Comment (merceyz) [reply package.json]: It's been over a week now / a few patch releases and the CI is happy so Vitest v2 it is.
